msprime/simulations.py

```diff
@@ -460,13 +465,23 @@ def simulate(
         underlying object may be used for every TreeSequence
         returned which will most likely lead to unexpected behaviour.
     """
+    argspec = inspect.getargvalues(inspect.currentframe())
```

Comment:
In the zillions of small replicates case, is it worthwhile skipping this step when record_provenance is False?

Reply:
I think this only gets called once for each call to `simulate`, not per replicate - but you're right there is no point calling it. Will move.